src/Facility.Definition/ServiceInfo.cs

```diff
@@ -155,9 +165,85 @@ ServiceMemberInfo DoExcludeTag(ServiceMemberInfo member)
 			}
 		}
 
+		private static void EnsureProperValidateUsage(ServiceTypeInfo type, ServiceElementWithAttributesInfo field)
+		{
+			var validateAttributes = field.GetAttributes("validate");
+
+			if (!validateAttributes.Any()) return;
+
+			if (validateAttributes.Count > 1)
+				field.AddValidationError(ServiceDefinitionUtility.CreateDuplicateAttributeError(validateAttributes[1]));
+
+			var validateAttribute = validateAttributes[0];
+
+			switch (type.Kind)
+			{
+				case ServiceTypeKind.Enum:
+					foreach (var validateAttributeParameter in validateAttribute.Parameters)
+						field.AddValidationError(ServiceDefinitionUtility.CreateUnexpectedAttributeParameterError(validateAttribute.Name, validateAttributeParameter));
+					break;
+
+				case ServiceTypeKind.String:
+				{
+					foreach (var validateAttributeParameter in validateAttribute.Parameters.Where(x => x.Name != "length" && x.Name != "pattern"))
+						field.AddValidationError(ServiceDefinitionUtility.CreateUnexpectedAttributeParameterError(validateAttribute.Name, validateAttributeParameter));
+
+					var lengthParameter = validateAttribute.TryGetParameter("length");
+					var patternParameter = validateAttribute.TryGetParameter("pattern");
```

Comment:
IMHO, `regex` is a better name than `pattern` (despite my example and OpenAPI), as it makes perfectly clear that a regular expression is being used.